### PR TITLE
Improve collective lessons

### PR DESCRIPTION
**Pull request by mabraham**
Also simplify e.g. use of printf in my examples elsewhere

### CODE REVIEW

**Review by wikfeldt — APPROVED** [review]
looks very good to me

File: content/custom.py [hunk 1]
```diff
@@ -119,7 +126,7 @@
 .. |{function}| replace:: ``{function}``
 .. |term-{function}| raw:: html
 
-   <a class="reference internal" href="../quick-reference/#term-{function}"><span class="xref std std-term"><code class="docutils literal notranslate">{function}</code></span></a>
+   <a class="reference internal" href="../quick-reference/index.html#term-{function}"><span class="xref std std-term"><code class="docutils literal notranslate">{function}</code></span></a>
```
Comment (wikfeldt) [inline content/custom.py:129]: what's this for? links to glossary seem to work without it

**Comment by wikfeldt** [comment]
i approved too early, have this question on the custom.py change

File: content/custom.py (hunk 1 above)
Comment (mabraham) [reply content/custom.py:129]: My local builds produce working links if we do it this way. Shouldn't matter in principle though...

Comment (wikfeldt) [reply content/custom.py:129]: ok thanks, thought i should ask before merging

**Pull request merged**